plugins/aws/cli_provisioner.go

```diff
@@ -25,22 +26,36 @@ func (p awsCLIProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput,
 
 func stripAndReturnProfileFlag(args *[]string) (string, error) {
```

Comment:
This is somewhat of a matter of taste, but passing the args as a pointer makes it somewhat difficult to figure out what is happenign with them. Can we instead just return the modified slice? Three return values is also not super nice, but since the return value is different from the profile, it's not likely that the profile and args return value will get mxied up.